index.ts

```diff
@@ -0,0 +1,27 @@
+'use strict'
+
+import { Config } from './config'
+
+/**
+* Start the profiling agent
+* Currently unimplemented
+*
+* @param {object=} config - Profiler configuration
+*
+* @example
+* profiler.start();
+*/
+function start(projectConfig: Config) {
+  console.log("UNIMPLEMENTED");
+}
+
+module.exports = {
+  start: start 
+};
+
+// If the module was --require'd from the command line, start the agent
+if (module.parent && module.parent.id === 'internal/preload') {
```

Comment:
This looks magic to me. Is this common to have such check in Node.js modules?

Reply:
This is what cloud-trace does to allow the module to be started when required from the cli
(https://github.com/GoogleCloudPlatform/cloud-trace-nodejs/blob/master/index.ts#L179-L182).
I'm not sure how common this type of check would be, though.

cloud-debug doesn't seem to support users requiring it at run time.